### PR TITLE
Upgrade to evmone 0.9.1 and hera 0.6.0 in buildpack

### PR DESCRIPTION
**Pull request by axic**
Pulled out from #13532. Should be merged together with it.

### CODE REVIEW

**Comment by axic** [comment]
Depends on https://github.com/ewasm/hera/pull/572 getting finished.

**Comment by axic** [comment]
Added the Hera image.

File: scripts/docker/buildpack-deps/Dockerfile.ubuntu2004
Comment (cameel) [inline scripts/docker/buildpack-deps/Dockerfile.ubuntu2004:51]: What about the other two dockerfiles? They still have 0.8.2 hard-coded.

**Review by cameel — COMMENTED** [review]
Is this PR independent of other changes? I.e. can it be merged without bumping the EVMC API version?

**Comment by axic** [comment]
> Is this PR independent of other changes? I.e. can it be merged without bumping the EVMC API version?

Well it is independent assuming you don't use these buildpack versions. The EVMC ABI is breaking so they can't be used without the other PR.

File: scripts/docker/buildpack-deps/Dockerfile.ubuntu2004
Comment (axic) [reply scripts/docker/buildpack-deps/Dockerfile.ubuntu2004:51]: Which other two?

I think ossfuzz was missed, but that's not even using it? Will fix that.

Comment (bshastry) [reply scripts/docker/buildpack-deps/Dockerfile.ubuntu2004:51]: ossfuzz uses it to fuzz the abi coder (because coding correctness cannot be interpreted via the yul interpreter and needs a standalone evm client).

**Review by bshastry — COMMENTED** [review]
LGTM. Since the buildpack versions created by this PR will only be used after https://github.com/ethereum/solidity/pull/13532 is merged, it is better to keep this PR open until then to tally buildpack versions used in that PR.

**Comment by nikola-matic** [comment]
@axic `install_evmone.ps1` powershell script for Windows also pulls the `0.8.0` evmone, although this script doesn't seem to be used in any of the CI steps, so my assumption is that it's just a convenience script for building the compiler on Windows.
Should this also be bumped? @cameel

**Comment by cameel** [comment]
@nikola-matic The script is used by `t_win_soltest`.

But yeah, I think that the version should be bumped everywhere.

File: scripts/docker/buildpack-deps/Dockerfile.ubuntu2004
Comment (cameel) [reply scripts/docker/buildpack-deps/Dockerfile.ubuntu2004:51]: Yeah, I meant ossfuzz. I think we should update all images as long as they're in the repo.

Not sure where I see the other one. I must have mistakenly looked at one of the images already updated here.

I see this was addressed so marking this as resolved.

**Review by cameel — DISMISSED** [review]
@nikola-matic The scripts should be updated, but that's addressed by #13532. Here we're only concerned with rebuilting the Docker images.

**Comment by nikola-matic** [comment]
I've rebased against fresh develop to get this into a merge-able state. The PR is approved, however I see @bshastry would prefer for this to stay open until #13532 is close to being merged. With our current push to trim our PR queue, I'd be fine with merging this right away. @bshastry, @cameel, any counter points?

**Comment by axic** [comment]
> I've rebased against fresh develop to get this into a merge-able state.

Oh, that means now need to change the hashes in the other PR. These two PRs should only be merged/rebased at the same time, they are unfortunately not independent. The reason they are split is due to the way buildpack works.

**Comment by nikola-matic** [comment]
> > I've rebased against fresh develop to get this into a merge-able state.
> 
> Oh, that means now need to change the hashes in the other PR. These two PRs should only be merged/rebased at the same time, they are unfortunately not independent. The reason they are split is due to the way buildpack works.

Sure thing, although if this isn't merge-able, then the PR should be converted to a draft so it's clear to everyone. At least the rebase + force push dismissed @cameel's approval.

**Comment by nikola-matic** [comment]
I've converted this to a draft as it should be merged together with https://github.com/ethereum/solidity/pull/13532. @axic feel free to revert this to a review-ready PR if needed.

**Comment by cameel** [comment]
Well, it's not mergeable but it's ready so we don't want people to skip reviewing it. So perhaps `has dependencies` label would be better here. These two PRs have a cyclic dependency on each other :)

**Comment by nikola-matic** [comment]
> Well, it's not mergeable but it's ready so we don't want people to skip reviewing it. So perhaps `has dependencies` label would be better here. These two PRs have a cyclic dependency on each other :)

Yup, I though about adding the `has dependencies` label, but at the same time the circular dependency made me think otherwise. The PR that does depend on this however, is in fact a draft, so really no idea what to think about this :)

**Comment by cameel** [comment]
I mean, I don't consider it a draft because there's nothing more to do in it. It's done.

**Review by cameel — APPROVED** [review]

**Comment by axic** [comment]
@cameel the only risk is merging this and then due to some other package update the hashes are changing in circle will cause a failure due to incompatible EVMC ABIs. If we expect the other PR to be merged very soon then it is safe to merge this, otherwise not.

**Comment by nikola-matic** [comment]
> @cameel the only risk is merging this and then due to some other package update the hashes are changing in circle will cause a failure due to incompatible EVMC ABIs. If we expect the other PR to be merged very soon then it is safe to merge this, otherwise not.

So that's kinda the reason I converted it to a draft - mainly so that it blocks merging before the other PR is ready as well. Not sure what to do know. You decide @axic, you're the boss.

**Comment by cameel** [comment]
> If we expect the other PR to be merged very soon then it is safe to merge this, otherwise not.

Should be pretty soon - the only blocker was this gas cost change to investigate. I see you pushed new commits there. So is that resolved?

But I'd still keep this open and merge them both at the same time.

> So that's kinda the reason I converted it to a draft - mainly so that it blocks merging before the other PR is ready as well. Not sure what to do know. You decide @axic, you're the boss.

At this point it's moot since this PR is already approved so either way achieves the goal. But my point was more in general - if you want reviews on such a PR it's probably better not to have it marked as draft so `has dependencies` would work better :) That label is a warning that the PR cannot be "just" merged.

**Comment by ekpyron** [comment]
For the record, I'd also tend to keep these things drafts to prevent accidental merging, until both PRs are ready :-). Draft meaning "cannot be merged" here, which is true for either of them, regardless of reviews, unless both of them are ready.

**Comment by axic** [comment]
Ready for final review.

**Pull request merged**
